jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -204,6 +204,9 @@ def _update_mark_scale(self, *args):
             self.aperture.scale_factor = 1.0
         else:
             self.aperture.scale_factor = self.slice_wavelength/self.reference_wavelength
+
+        self.app._metadata['_pixel_scale_factor'] = self.aperture.scale_factor
```

Comment:
Is `app._metadata` a completely new thing? I don't remember ever seeing that before. Shouldn't this info be stored in the `data.metadata` instead? Surely this cannot be an app-wide setting?